### ISSUE TITLE
Search criteria for `co2dice tstamp recv` & `co2dice project trecv`

### ISSUE THREAD

**Issue by MaikeruJon** [JRCSTU/CO2MPAS-TA#20]
How long does it take before my project shows up in the response from `co2dice tstamp recv`?
I received the e-mail reply from the time stamper and thought that I could get the result of my latest submission by the command `co2dice tstamp recv` instead of `co2dice project tparse` and then copying and inserting. But half-an-hour after I received the reply (not the acknowledgement, the actual reply) from timestamper, I am just receiving a list of old timestamper mails.
Am I misunderstanding this command?

**Comment by VictorValverde** [comment]
Dear @MaikeruJon 
You are right, the intention of the command ```co2dice tstamp recv``` is to fetch tstamps from IMAP server and to derive the OK/SAMPLE decision. 
   
In case it does not work properly after having received the dice response email some options can be added to the command to facilitate _finding_ the proper dice response:

- ```co2dice tstamp recv --mailbox "Spam"``` will search messages in a particular folder of the email (Spam in the example). The default is Inbox.
- ```co2dice tstamp recv "IP-10-AAA-2017-1003"``` will search messages containing the specified familiy-id in the subject/body of the message.
- ```co2dice tstamp recv --before "08/01/17"``` will search messages sent before the specified day (1 st August 2017).
- ```co2dice tstamp recv --after "08/01/17"``` will search messages sent after the specified day (1 st August 2017).
- ```co2dice tstamp recv --on "08/01/17"``` will search messages sent on the specified day (1 st August 2017).

In case none of this additional options work, could you please send us in an e-mail the following command?
```co2dice tstamp recv -v```
Kind regards.

**Comment by ankostis** [comment]
To clarify further the situation, first some definitions:
- I assume that with "acknowledgement" you mean the reply-back email that is sent `From:` tstamper-address, with subject "Proof of posting ...", correct?
- And the "actual reply" are the stamped-emails that are sent `From:` your email-address to the `tstamp_recepient` addresses, with subject containing just the subject-prefix and *project-id` correct?

Dice comes pre-populated with some default criteria so that `(t)recv` commands fetch ONLY "acknowledgement" emails, because that's the typical use-case.  In the end, the resulting criteria are those printed as logging statement shown below:
```
```
The parameters controlling those *default* criteria to are the following (for `v1.5.7`): 

```python
TstampReceiver.subject_prefix
TstampReceiver.email_criteria  ## This param will be renamed as `rfc_criteria` in forthcoming release 1.6.x
```

Tip: you may view them with `co2dice config desc criteria` command.  And in 1.5.x you view the actual values with `co2dice config show TstampReceiver`.

In order to fetch emails received from the `tstamp_recipient` addresses you have to override these defaults.

To summarize things up:
- If you want to parse the acknowledgment "proof of posting' reply-back email, you have to connect to your sending server and let the default criteria as is.
- If you want to parse the tstamps in other accounts, you have to connect to those accounts, and override the default criteria.  If that is what you ask, please send us your `co2dice_config.py` file to our through email, and specify from which account you want to parse tstamps, so that we can modify them.

**Comment by MaikeruJon** [comment]
Many thanks to the team for your typically prompt and helpful replies. I have had success and I now feed back as follows:
1) "I assume that with "acknowledgement" you mean..." Yes, that is what I meant.
2) "And the "actual reply" are the stamped-emails..." Yes, that is what I meant
3) "co2dice tstamp recv "IP-10-AAA-2017-1003" will search messages containing the specified familiy-id in the subject/body of the message." Thank you. This worked for me. 
[tstamp recv history.txt](https://github.com/JRCSTU/CO2MPAS-TA/files/1195798/tstamp.recv.history.txt)
Please refer to the uploaded file that shows the history.
[co2dice_config.py.txt](https://github.com/JRCSTU/CO2MPAS-TA/files/1195801/co2dice_config.py.txt)
I think that there is noting so confidential in it, so I am uploading my configuration file for reference.
Thanks again.
